docs/development/new_component_onboarding.md

```diff
@@ -0,0 +1,71 @@
+# Onboarding a New Component into TheRock  
+*(For new component teams — use this as your checklist and reference)*
+
+## Purpose  
+This document describes the steps for bringing a new component into TheRock (ROCm build platform) from initial commit through CI integration, using the example of a previous component add-up.
+
+## Pre-Requisites  
+Before you start, ensure you have:  
+- A publicly accessible upstream repository for your component (cloneable).  
+- A reproducible build (preferably CMake) of your component.  
+- A test suite (smoke and full/regression) that can be run non-interactively in CI.  
+- Any patches needed for TheRock integration (packaged as `.patch` files).  
+- Defined artifact outputs (what binaries or libraries will be produced).  
+- A point of contact in your team for component maintenance.
+
+## Step-by-Step Process  
+
+### 1. Add the source into TheRock  
+- Add your component under `math-libs/<component-name>` (or appropriate category) as a submodule or source directory.  
+- Update `.gitmodules` accordingly if using submodules.  
+- Commit with message: `Add <component-name> submodule`.
```

Comment:
This duplicates some instructions from https://github.com/ROCm/TheRock/blob/main/docs/development/git_chores.md#adding-a-new-submodule. Could link to that page here.

(But agreed with the other comment about using an existing superrepo instead of adding a new submodule)